### ISSUE TITLE
"TablePlus only allows editing the value queried from a single table"

### ISSUE THREAD

**Issue by LeonardChallis** [TablePlus/TablePlus#1840]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

Server version: 5.7.29-32-log Percona Server (GPL), Release 32, Revision 56bce88

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

Version 3.4.0 (304)

3. The steps to reproduce this issue:

Run a query to a single table, then try to edit a column.

![tabeplus](https://user-images.githubusercontent.com/404920/80585072-97d75880-8a0a-11ea-8d98-e1b9ca3142e2.gif)
(sorry for the quality/speed of the gif - used an online converter)

This seems to happen on *any* set of data queried from a single table.

**Comment by huyphams** [comment]
Did you run the query `USE <some_db>;` and then run the select query? I don't see the shipping_rate table in the auto-suggestion which indicates that the database was not opened, and TablePlus could not found it in the current context so that's why it won't allow you to edit it (it does not know the table name and database so it could not generate the query).

If you run the query like this: `SELECT * FROM db1.test;` or open that database before running any query, you will be able to edit the content.

**Comment by LeonardChallis** [comment]
> Did you run the query `USE <some_db>;` and then run the select query? I don't see the shipping_rate table in the auto-suggestion which indicates that the database was not opened, and TablePlus could not find it in the current context so that's why it wouldn't allow you to edit the row (it didn't know the table name and database so it could not generate the query).
> 
> If you run the query like this: `SELECT * FROM db1.test;` or open that database before running any query, you will be able to edit the content.

Yes, I did. When I run `use any_schema;` it returns `Query 1 OK: , 0 rows affected`. Then I run `SELECT * FROM any_relation;` and then try edit one of the columns and I get that exact same message.

I can confirm that doing `SELECT * FROM any_schema.any_relation;` *does* allow me to edit the columns. Am I missing something obvious here?

**Comment by huyphams** [comment]
TablePlus will not fetch the items of the database until you write it down or open it (the reason is performance, there are unpredictable a number of schemas). For example, if you open the database and all the items are listed on the left sidebar, TablePlus will allow you to edit the result. If you don't open the database but you write it in the query, TablePlus also knows that you want to query its items and will allow you to edit the result (it smarts enough to fetch the database you listed in the queries).
